lib/src/Uncertainty/Distribution/openturns/PointConditionalDistribution.hxx

```diff
@@ -0,0 +1,219 @@
+  Bool isDiscrete() const override;
+
+  /** Check if the distribution is integral */
+  Bool isIntegral() const override;
+
+  /** Get the support of a discrete distribution that intersect a given interval */
+  using DistributionImplementation::getSupport;
+  Sample getSupport(const Interval & interval) const override;
+
+  /** Get the standard deviation of the distribution */
+  Point getStandardDeviation() const override;
+
+  /** Get the skewness of the distribution */
+  Point getSkewness() const override;
+
+  /** Get the kurtosis of the distribution */
+  Point getKurtosis() const override;
+
+  /** Parameters value accessors */
+  void setParameter(const Point & parameter) override;
+  Point getParameter() const override;
+
+  /** Parameters description accessor */
+  Description getParameterDescription() const override;
+
+  /** Get the PDF gradient of the distribution */
+  using DistributionImplementation::computePDFGradient;
+  Point computePDFGradient(const Point & point) const override;
+
+  /** Get the CDF gradient of the distribution */
+  using DistributionImplementation::computeCDFGradient;
+  Point computeCDFGradient(const Point & point) const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+  /* Interface specific to PointConditionalDistribution */
+
+  /** Conditioned distribution accessor */
+  Distribution getDistribution() const;
+
+  /** Probability table accessor */
+  Indices getConditioningIndices() const;
+  Point getConditioningValues() const;
+
+  /** Get the simplified version (or clone the distribution) */
+  Distribution getSimplifiedVersion() const;
+
+protected:
+
+private:
+  /** Expand the given marginal point to the underlying distribution argument point */
+  Point expandPoint(const Point & point) const;
+
+  /* Get the simplified version */
+  Bool hasSimplifiedVersion(Distribution & simplified) const;
+
+  // decompose elliptical distribution mu/cov
+  Point decompose(CovarianceMatrix & C) const;
+
+  /** Compute the mean of the distribution */
+  void computeMean() const override;
+
+  /** Compute the covariance of the distribution */
+  void computeCovariance() const override;
+
+  /** Compute the numerical range of the distribution given the parameters values */
+  void computeRange() override;
+
+  /** Update byproducts */
+  void update();
+
+  // main attributes
+  Distribution distribution_;
+  Indices conditioningIndices_;
+  Point conditioningValues_;
+
+  // cache some byproducts
+  Indices nonConditioningIndices_;
+  Distribution marginalConditionedDistribution_;
+  Scalar logNormalizationFactor_ = 0.0;
+
+  // simplified instance
+  Distribution simplifiedVersion_;
+  Bool useSimplifiedVersion_ = false;
+
+  // for CDF integration
+  IntegrationAlgorithm integrationAlgorithm_;
```

Comment:
It seems to me that being able to update the `integrationAlgorithm_` with a `setIntegrationAlgorithm()` method would be beneficial. This is because `IteratedQuadrature` will not always be able to integrate this type of problem, given that the number of nodes generated by the algorithm grows relatively fast with the dimension. When `ExperimentIntegration` will inherit from `IntegrationAlgorithm` (after #2709 is fixed), then the user will be able to use a sampling based integration method, which will allow to manage problems in higher dimensions.